sklearn/decomposition/_nmf.py

```diff
@@ -1717,3 +1752,632 @@ def inverse_transform(self, W):
+    large scale nonnegative matrix and tensor factorizations."
+    IEICE transactions on fundamentals of electronics, communications and
+    computer sciences 92.3: 708-721, 2009.
+
+    Fevotte, C., & Idier, J. (2011). Algorithms for nonnegative matrix
+    factorization with the beta-divergence. Neural Computation, 23(9).
+
+    Lefevre, A., Bach, F., Fevotte, C. (2011). Online algorithms for
+    nonnegative matrix factorization with the Itakura-Saito divergence.
+    WASPA (https://doi.org/10.1109/ASPAA.2011.6082314,
+    https://hal.archives-ouvertes.fr/hal-00602050)
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> X = np.array([[1, 1], [2, 1], [3, 1.2], [4, 1], [5, 0.8], [6, 1]])
+    >>> from sklearn.decomposition import MiniBatchNMF
+    >>> model = MiniBatchNMF(n_components=2, init='random', random_state=0)
+    >>> W = model.fit_transform(X)
+    >>> H = model.components_
+    """
+
+    def __init__(
+        self,
+        n_components=None,
+        *,
+        init=None,
+        batch_size=1024,
+        solver="mu",
+        beta_loss="frobenius",
+        tol=1e-4,
+        max_no_improvement=10,
+        max_iter=200,
+        alpha_W=0.0,
+        alpha_H="same",
+        l1_ratio=0.0,
+        forget_factor=0.7,
+        fresh_restarts=False,
+        fresh_restarts_max_iter=30,
+        transform_max_iter=None,
+        random_state=None,
+        verbose=0,
+    ):
+
+        super().__init__(
+            n_components=n_components,
+            init=init,
+            solver=solver,
+            beta_loss=beta_loss,
+            tol=tol,
+            max_iter=max_iter,
+            random_state=random_state,
+            alpha_W=alpha_W,
+            alpha_H=alpha_H,
+            l1_ratio=l1_ratio,
+            verbose=verbose,
+        )
+
+        self.max_no_improvement = max_no_improvement
+        self.batch_size = batch_size
+        self.forget_factor = forget_factor
+        self.fresh_restarts = fresh_restarts
+        self.fresh_restarts_max_iter = fresh_restarts_max_iter
+        self.transform_max_iter = transform_max_iter
+
+    def _check_params(self, X):
+        super()._check_params(X)
+
+        # solver
+        if not isinstance(self.solver, str) or self.solver != "mu":
+            raise ValueError(
+                f"Invalid solver parameter '{self.solver}'. "
+                "Only solver='mu' is accepted."
+            )
+
+        # batch_size
+        self._batch_size = self.batch_size
+        if not isinstance(self._batch_size, numbers.Integral) or self._batch_size <= 0:
+            raise ValueError(
+                "batch_size must be a positive integer, got "
+                f"{self._batch_size!r} instead."
+            )
+        self._batch_size = min(self._batch_size, X.shape[0])
+
+        # forget_factor
+        # TODO
+        self._rho = self.forget_factor ** (self._batch_size / X.shape[0])
```

Comment:
What is to be done here?

Reply:
I'm not found of this definition but that's the definition in the ref paper. I think we can let it as is for now and if users complain that it's not easy to tune we can adapt later. I removed the comment